### PR TITLE
gizmo tool rework

### PR DESCRIPTION
**Pull request by cblech**
Fixes #861m8e65n, #861m41y69, and #861m41zwt

### CODE REVIEW

**Comment by MGH-Automator** [comment]
Task linked: [CU-861m8e65n Fix bug translation when parent of parent scaled](https://app.clickup.com/t/861m8e65n)

**Comment by MGH-Automator** [comment]
Task linked: [CU-861m41y69 global and local edditing for gizmo tools](https://app.clickup.com/t/861m41y69)

**Comment by MGH-Automator** [comment]
Task linked: [CU-861m41zwt Implement snapping](https://app.clickup.com/t/861m41zwt)

File: Assets/Scripts/EditorState/GizmoState.cs [hunk 1]
```diff
@@ -1,37 +1,156 @@
...
         {
-            Translate,
-            Rotate,
-            Scale
+            OnlyPrimaryAxis,
+            EntirePlane
         }
 
-        private Mode _currentMode;
+        // Dependencies
```
Comment (tcdrop) [inline Assets/Scripts/EditorState/GizmoState.cs]: This comment can be removed.

**Review by tcdrop — CHANGES_REQUESTED** [review]

**Comment by MichiGandi** [comment]
1) Drag an asset out of the Asset browser
The new object is now selected and Gizmo is visable (good)
2) Press Undo
The object is now gone, but the gizmo is still there (not good)

**Comment by MichiGandi** [comment]
Should the Tmp Area be removed?
Maybe create one Area "Gizmo Settings".
grafik

**Comment by MichiGandi** [comment]
I could not find out how to use global mode Gizmos.

**Comment by tcdrop** [comment]
When translating an object with snapping enabled the position doesn't snap correctly.

Test values: 
Gizmo tool does snap: 1
Gizmo tool translate snapping: 1
Object starting position: 0,0,16.4

Expected result: The object should snap to z-coordinate 15,14,13,12 etc.
Actual result: The object snaps to z-coordinate 15.5999, 12.3999 etc.

**Comment by cblech** [comment]
> 1. Drag an asset out of the Asset browser
>        The new object is now selected and Gizmo is visable (good)
> 
>     2. Press Undo
>        The object is now gone, but the gizmo is still there (not good)

This is a problem with the asset browser and should be fixed in a separate PR

**Pull request merged**